chips/sam4l/src/usart.rs

```diff
@@ -725,24 +920,21 @@ impl hil::spi::SpiMaster for USART {
     type ChipSelect = Option<&'static hil::gpio::Pin>;
 
     fn init(&self) {
-        let regs: &USARTRegisters = unsafe { &*self.registers };
+        let regs: &UsartRegisters = unsafe { &*self.registers };
 
         self.usart_mode.set(UsartMode::Spi);
         self.enable_clock();
 
         // Set baud rate, default to 2 MHz.
         self.set_baud_rate(2000000);
 
-        let mode =
-            0xe << 0 /* SPI Master mode */
-            | 0 << 4 /* USCLKS*/
-            | 0x3 << 6 /* Character Length 8 bits */
-            | 0x4 << 9 /* No Parity */
-            | 1 << 18 /* USART drives the clock pin */;
-        self.set_mode(mode);
+        regs.mr.write(
+            Mode::MODE::SPI_MASTER + Mode::USCLKS::CLK_USART + Mode::CHRL::BITS8 + Mode::PAR::NONE
+                + Mode::CLKO::SET,
+        );
 
         // Disable transmitter timeguard
-        regs.ttgr.set(4);
+        regs.ttgr.write(TxTimeGuard::TG.val(4));
```

Comment:
Okay, this isn't your fault, but could you update the comment here? That is not equivalent to disabling it per the manual (which would be 0). This 4 means we have four bit periods of guard before RTS/CTS toggle after a message. Which feels fine to me. Clearly someone thought it was necessary, probably for nRF serialization.